tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
+    """
 
-    @pytest.mark.parametrize("grouping_strategy", [None, "default", "qwc", "wires"])
-    @pytest.mark.parametrize("shots", [None, 20000, [20000, 30000, 40000]])
+    @pytest.mark.parametrize("grouping_strategy", [None, "qwc", "wires"])
+    @pytest.mark.parametrize("shots", [None, 10, [10, 20, 30]])
     @pytest.mark.parametrize(
-        "params, expected_results",
+        "params",
         [
-            (
-                [np.pi / 4, 3 * np.pi / 4],
-                [
-                    0.5,
-                    -0.5 * np.cos(np.pi / 4),
-                    0.5 + np.cos(np.pi / 4) * 0.5 + 2.0 * 0.5 + 1,
-                    np.dot(
-                        [0.1, 0.2, 0.3, 0.4],
-                        [-0.5, -0.5 * np.cos(np.pi / 4), 0.5 * np.cos(np.pi / 4), 1],
-                    ),
-                    1.5,
-                ],
-            ),
-            (
-                [[np.pi / 4, 3 * np.pi / 4], [3 * np.pi / 4, 3 * np.pi / 4]],
-                [
-                    [0.5, -0.5],
-                    [-0.5 * np.cos(np.pi / 4), -0.5 * np.cos(np.pi / 4)],
-                    [
-                        0.5 + np.cos(np.pi / 4) * 0.5 + 2.0 * 0.5 + 1,
-                        -0.5 - np.cos(np.pi / 4) * 0.5 - 2.0 * 0.5 + 1,
-                    ],
-                    [
-                        np.dot(
-                            [0.1, 0.2, 0.3, 0.4],
-                            [-0.5, -0.5 * np.cos(np.pi / 4), 0.5 * np.cos(np.pi / 4), 1],
-                        ),
-                        np.dot(
-                            [0.1, 0.2, 0.3, 0.4],
-                            [0.5, 0.5 * np.cos(np.pi / 4), -0.5 * np.cos(np.pi / 4), 1],
-                        ),
-                    ],
-                    [1.5, 1.5],
-                ],
-            ),
+            [0.1, 0.2, 0.3],
+            [[0.1, 0.1], [0.2, 0.2], [0.3, 0.3]],
         ],
     )
-    def test_multiple_expval(self, grouping_strategy, shots, params, expected_results, seed):
-        """Tests that a QNode with multiple expval measurements is executed correctly"""
+    def test_measurement_of_single_hamiltonian(self, grouping_strategy, shots, params):
+        """Tests executing a QNode returning a single measurement of a Hamiltonian."""
 
-        dev = qml.device("default.qubit", wires=2, shots=shots, seed=seed)
+        coeffs, obs = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6], single_term_obs_list
 
-        obs_list = complex_obs_list
+        dev = qml.device("default.qubit", wires=3, shots=shots)
 
+        @partial(split_non_commuting, grouping_strategy=grouping_strategy)
         @qml.qnode(dev)
         def circuit(angles):
             qml.RX(angles[0], wires=0)
-            qml.RY(angles[1], wires=0)
-            qml.RX(angles[0], wires=1)
             qml.RY(angles[1], wires=1)
-            return [qml.expval(obs) for obs in obs_list]
+            qml.RZ(angles[2], wires=2)
+            return qml.expval(qml.Hamiltonian(coeffs, obs))
```

Comment:
Do we need an execution for this validation?

Reply:
These are integration tests of applying the transform on a QNode and making sure that it executes correctly so I would say yes?

Reply:
I guess this is something we can mock out the device with, as it doesn't need the device end-to-end, just the QN integration, right? Though, I'd fine with making a note/follow up story to do this and proceed as is for now.

Reply:
The test was marked `pytest.mark.system`. I guess I shouldn't have named it `TestQNodeIntegration`. I meant to test the full execution pipeline, because we want to be testing with real result batches returned from device execution of the tapes that we generated. I wrote the original test, and I believe these types of tests did reveal certain edge cases that wasn't considered before.

Reply:
Let's make note of this separation to happen as a future TODO, and plan to reduce these testing overheads. I still see system level tests here as non essential, and to be brought to unit/integration level.
If you are happy with that assessment and we can capture those items for future work, I'm happy to approve as is.

Reply:
I think it depends on the transform. For some transforms where nothing complicated happens in post-processing, this is non-essential, but the `split_non_commuting` contains complicated post-processing of the execution results. It is technically possible to test the post processing transform with mocked device execution, but given the complexity of the return type specification (5 dimensions, each of them may or may not have been squeezed out in different scenarios), I'm not confident that the mock is able to exactly mimic the behaviour of actual execution, and it's probably not worth it to do so. For these tests I have set the number of shots to `10` to avoid too much overhead, since we're not testing for numerical correctness here, just that the circuit executes without error and the returned values have the correct shapes.

Reply:
I think we need to disagree and commit on this @astralcai :) 
Let's track this item for future work so we can get this PR through.

Reply:
Sounds good. I've created a story on SC and tagged you in it. @mlxd

Reply:
Awesome, TY.